src/Controller/AdminController.php

```diff
@@ -1038,15 +1034,45 @@ public function tagsAction(Request $request, TagManager $tagManager)
     /**
      * @Route("/admin/deliveries/pricing", name="admin_deliveries_pricing")
      */
-    public function pricingRuleSetsAction()
+    public function pricingRuleSetsAction(Request $request, PaginatorInterface $paginator, PricingRuleSetManager $pricingRuleSetManager, ApplicationsNormalizer $normalizer)
     {
         $this->denyAccessUnlessGranted('ROLE_ADMIN');
-        $ruleSets = $this->getDoctrine()
-            ->getRepository(Delivery\PricingRuleSet::class)
-            ->findAll();
-        return $this->render('admin/pricing.html.twig', [
-            'ruleSets' => $ruleSets
-        ]);
+
+        // the way we get the results for applications of pricing rule set is not very efficient (3 requests per rule set), so let's not load too much of them
+        // if needed optimize (and complexifiy !) the query to get applications of the pricing rule set
+        $qb = $this->getDoctrine()->getRepository(Delivery\PricingRuleSet::class)
+            ->createQueryBuilder('rs')
+            ->orderBy('rs.name', 'ASC')
+            ->setFirstResult(($request->query->getInt('p', 1) - 1) * self::ITEMS_PER_PAGE / 2)
```

Comment:
This seems fragile, i'll add `max()` function to be sure the value cannot go under 0.
```suggestion
            ->setFirstResult(max(($request->query->getInt('p', 1) - 1),0) * self::ITEMS_PER_PAGE / 2)
```